src/openms_gui/source/VISUAL/TopDownViewWidget.cpp

```diff
@@ -0,0 +1,375 @@
+
+      // rt
+      addDoubleItemToBottomRow_(spectrum.getRT(), 2, c);
+
+
+      // fill precursor information in columns
+      if (!precursors.empty())
+      {
+        const Precursor & first_precursor = precursors.front();
+
+        // set precursor m/z
+        item = table_widget_->itemPrototype()->clone();
+        item->setData(Qt::DisplayRole, first_precursor.getMZ());
+        item->setBackgroundColor(c);
+        item->setTextColor(Qt::blue); // draw precursor information in blue
+        table_widget_->setItem(table_widget_->rowCount() - 1, 3, item); // precursor m/z
+
+        // set charge
+        item = table_widget_->itemPrototype()->clone();
+        item->setBackgroundColor(c);
+        item->setData(Qt::DisplayRole, first_precursor.getCharge());
+        table_widget_->setItem(table_widget_->rowCount() - 1, 4, item); // precursor charge
+      }
+      else
+      { // has no precursor
+        addTextItemToBottomRow_("-", 3, c); // precursor m/z
+        addTextItemToBottomRow_("-", 4, c); // precursor charge
+      }
+
+      if (i == layer_->getCurrentSpectrumIndex())
+      {
+        selected_row = item->row(); // get model index of selected spectrum
+      }
+    }
+
+    table_widget_->setSortingEnabled(true);
+    table_widget_->setHorizontalHeaderLabels(header_labels);
+    table_widget_->resizeColumnsToContents();
+
+    if (selected_row != -1)  // select and scroll down to item
+    {
+      table_widget_->selectRow(selected_row);
+      QTableWidgetItem* selected_item = table_widget_->item(selected_row, 0);
+      selected_item->setSelected(true);
+      table_widget_->setCurrentItem(selected_item);
+      table_widget_->scrollToItem(selected_item);
+    }
+
+    table_widget_->blockSignals(false);
+    table_widget_->setUpdatesEnabled(true);
+    has_data_ = true;
+  }
+
+  void TopDownViewWidget::headerContextMenu_(const QPoint& pos)
+  {
+    // create menu
+    QMenu context_menu(table_widget_);
+
+    // extract header labels
+    QStringList header_labels;
+    for (int i = 0; i != table_widget_->columnCount(); ++i)
+    {
+      QTableWidgetItem* ti = table_widget_->horizontalHeaderItem(i);
+      if (ti != nullptr)
+      {
+        header_labels.append(ti->text());
+      }
+    }
+
+    // add actions which show/hide columns
+    for (int i = 0; i != table_widget_->columnCount(); ++i)
+    {
+      QTableWidgetItem* ti = table_widget_->horizontalHeaderItem(i);
+      if (ti == nullptr) continue;
+      QAction* action = context_menu.addAction(ti->text(), [=]() {
+              // invert visibility upon clicking the item
+              table_widget_->setColumnHidden(i, !table_widget_->isColumnHidden(i));
+        });
+      action->setCheckable(true);
+      action->setChecked(!table_widget_->isColumnHidden(i));
+    }
+    context_menu.exec(table_widget_->mapToGlobal(pos));
+  }
+
+  void TopDownViewWidget::addTextItemToBottomRow_(const QString& text, Size column_index, const QColor& c)
```

Comment:
this code is copy&paste from other IDWidgets
Subclass QTableWidget and add the methods there?!